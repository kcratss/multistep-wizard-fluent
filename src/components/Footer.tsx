import * as React from "react";
import { DefaultButton, PrimaryButton } from "@fluentui/react/lib/Button";
import { Stack } from "@fluentui/react/lib/Stack";
import { NavContext } from "../context/NavContext";

export type IFooterProps = {
  shouldDisablePrev?: () => boolean;
  shouldDisableNext?: () => boolean;
  submitText?: string;
  onSubmit?: () => void;
  navStateCallback: (data: INavDetails) => void;
};

export type INavDetails = {
  currentPageIndex: number;
}

export const Footer = (props: IFooterProps) => {
  const { stepDetails, setStepDetails, stepList } = React.useContext(
    NavContext
  );

  return (
    <footer>
      <Stack horizontal horizontalAlign="space-between">

      {stepDetails.currentPageIndex !== stepList.length - 1 && (
        <Stack.Item>
          
            <PrimaryButton
              onClick={() => {
                setStepDetails({
                  currentPageIndex: stepDetails.currentPageIndex + 1,
                });
                const data: INavDetails = {
                  currentPageIndex: stepDetails.currentPageIndex + 1
                }
                props.navStateCallback(data);
              }}
              disabled={
                props.shouldDisableNext ? props.shouldDisableNext() : false
              }
            >
              Next
            </PrimaryButton>
          
        </Stack.Item>
)}

        {stepDetails.currentPageIndex === stepList.length - 1 && (
        <Stack.Item>
          
            <PrimaryButton
              onClick={props.onSubmit}
              disabled={
                props.shouldDisableNext ? props.shouldDisableNext() : false
              }
            >
              {props.submitText?.length !== 0 ? props.submitText : "Submit"}
            </PrimaryButton>          
        </Stack.Item>
        )}

{stepDetails.currentPageIndex !== 0 && (
        <Stack.Item>
          
            <DefaultButton
              onClick={() => {
                setStepDetails({
                  currentPageIndex: stepDetails.currentPageIndex - 1,
                });
                const data: INavDetails = {
                  currentPageIndex: stepDetails.currentPageIndex - 1
                }
                props.navStateCallback(data);
              }}
              disabled={
                props.shouldDisablePrev ? props.shouldDisablePrev() : false
              }
            >
              Back
            </DefaultButton>
         
        </Stack.Item>
 )}
      </Stack>
    </footer>
  );
};
