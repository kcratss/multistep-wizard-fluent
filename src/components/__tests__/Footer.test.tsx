import React from "react";
import "@testing-library/jest-dom";
import { Footer } from "../Footer";
import { render } from "@testing-library/react";

describe("Footer tests", () => {
  test("basic snapshot", () => {
    const { asFragment } = render(
      <Footer
        shouldDisableNext={() => {
          return false;
        }}
        shouldDisablePrev={() => {
          return false;
        }}
      ></Footer>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
