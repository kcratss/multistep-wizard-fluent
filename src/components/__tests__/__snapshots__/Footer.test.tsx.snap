// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests basic snapshot 1`] = `
<DocumentFragment>
  <footer
    style="margin-left: 23vw;"
  >
    <div
      class="ms-Stack css-40"
    >
      <div
        class="ms-StackItem css-41"
      />
      <div
        class="ms-StackItem css-41"
      >
        <button
          class="ms-Button ms-Button--primary root-42"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-43"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-44"
            >
              <span
                class="ms-Button-label label-46"
                id="id__0"
              >
                Next
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
